changelog/fix16997.dd

```diff
@@ -0,0 +1,23 @@
+-preview=intpromote switch has now been set to true by default now.
+
+use -revert=intpromote to revert the changes
+
+integer promotions now follow C integral promotions rules consistently 
+
+It affects operands of type byte, ubyte, short, ushort, char, and wchar. The operands are promoted to int before the operator is applied. The result type will now be int.
+
+The values computed will be different for some values of the operands:
+
+    All values of - and ~ applied to ubyte, ushort, char, and dchar operands will change.
+
+    The values of -byte(-128) and -short(-32768) will change.
+
+If one of the '-transition=' switches is not specified, a deprecation will be emitted for these operations. To fix the deprecation and work as desired with or without the '-transition=' switch:
```

Comment:
The transition option is legacy and deliberately undocumented.

Reply:
Removed and re-written.